src/server/pfs/server/driver_fsck.go

```diff
@@ -317,26 +318,21 @@ func fsckCommits(commitInfos map[string]*pfs.CommitInfo, onError func(error) err
 	return nil
 }
 
-func (d *driver) listReferencedCommits(ctx context.Context) (map[string]*pfs.Commit, error) {
+func (d *driver) listReferencedCommits(tx *pachsql.Tx) (map[string]*pfs.Commit, error) {
 	cs := make(map[string]*pfs.Commit)
-	if err := dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, sqlTx *pachsql.Tx) error {
-		var ids []string
-		if err := sqlTx.Select(&ids, `SELECT commit_id from  pfs.commit_totals`); err != nil {
-			return errors.Wrap(err, "select commit ids from pfs.commit_totals")
-		}
-		for _, id := range ids {
-			cs[id] = pfsdb.ParseCommit(id)
-		}
-		ids = make([]string, 0)
-		if err := sqlTx.Select(&ids, `SELECT commit_id from  pfs.commit_diffs`); err != nil {
-			return errors.Wrap(err, "select commit ids from pfs.commit_diffs")
-		}
-		for _, id := range ids {
-			cs[id] = pfsdb.ParseCommit(id)
-		}
-		return nil
-	}); err != nil {
-		return nil, err
+	var ids []string
```

Comment:
Does this need to be in a transaction now?  What was wrong with opening a new transaction, like the old code?

Reply:
Based on testing that @albscui and I did, it seemed to be leading to dangling commit scenarios, but we are not 100% sure why. Our theory was that transactions were being interleaved and causing a state mismatch.

Reply:
update: I split this across 3 transactions (list repos, list matching commits & branches, and listReferencedCommits)